### PR TITLE
Kernel: Propagate errors of HIDDevices

### PR DESCRIPTION
**Pull request by L0ric0**
This PR ads the plumbing to get the errors encountered in HIDDevices out to be handled.

The error handling itself was not changed.

If someone knows more appropriate error codes instead of `EIO` please tell me.

### CODE REVIEW

File: Kernel/Devices/HID/PS2MouseDevice.cpp [hunk 1]
```diff
@@ -139,70 +139,69 @@ MousePacket PS2MouseDevice::parse_data_packet(const RawPacket& raw_packet)
...
-u8 PS2MouseDevice::send_command(u8 command)
+ErrorOr<u8> PS2MouseDevice::send_command(u8 command)
 {
-    u8 response = m_i8042_controller->send_command(instrument_type(), command);
-    if (response != I8042Response::Acknowledge)
+    u8 response = TRY(m_i8042_controller->send_command(instrument_type(), command));
+
+    if (response != I8042Response::Acknowledge) {
         dbgln("PS2MouseDevice: Command {} got {} but expected ack: {}", command, response, static_cast<u8>(I8042Response::Acknowledge));
+        // FIXME: Is this the correct errno value for this?
```
Comment (supercomputer7) [inline Kernel/Devices/HID/PS2MouseDevice.cpp:159]: It's probably the right errno, but let other developers to look at this and to check this.

File: Kernel/Devices/HID/PS2MouseDevice.cpp [hunk 2]
```diff
@@ -139,70 +139,69 @@ MousePacket PS2MouseDevice::parse_data_packet(const RawPacket& raw_packet)
...
 
-u8 PS2MouseDevice::send_command(u8 command, u8 data)
+ErrorOr<u8> PS2MouseDevice::send_command(u8 command, u8 data)
 {
-    u8 response = m_i8042_controller->send_command(instrument_type(), command, data);
-    if (response != I8042Response::Acknowledge)
+    u8 response = TRY(m_i8042_controller->send_command(instrument_type(), command, data));
+    if (response != I8042Response::Acknowledge) {
         dbgln("PS2MouseDevice: Command {} got {} but expected ack: {}", command, response, static_cast<u8>(I8042Response::Acknowledge));
+        // FIXME: Is this the correct errno value for this?
```
Comment (supercomputer7) [inline Kernel/Devices/HID/PS2MouseDevice.cpp:170]: It's probably the right errno, but let other developers to look at this and to check this.

**Review by bgianfo — APPROVED** [review]
Looks reasonable to me!

**Pull request merged**